src/processor/operator/persistent/reader/csv/csv_error_handler.cpp

```diff
@@ -228,27 +189,39 @@ void LocalCSVFileErrorHandler::handleError(BaseCSVReader* reader, const CSVError
         flushCachedErrors();
     }
 
-    ++linesPerBlock[error.blockIdx].invalidLines;
-    cachedErrors.push_back(error);
+    ++linesPerBlock[error.warningData.blockIdx].invalidLines;
+    if (cacheIgnoredErrors) {
+        cachedErrors.push_back(error);
+    }
 }
 
-void LocalCSVFileErrorHandler::reportFinishedBlock(BaseCSVReader* reader, uint64_t blockIdx,
-    uint64_t numRowsRead) {
-    sharedErrorHandler->reportFinishedBlock(reader, blockIdx, numRowsRead);
+void LocalCSVFileErrorHandler::reportFinishedBlock(uint64_t blockIdx, uint64_t numRowsRead) {
+    linesPerBlock[blockIdx].validLines += numRowsRead;
+    linesPerBlock[blockIdx].doneParsingBlock = true;
 }
 
 void LocalCSVFileErrorHandler::setHeaderNumRows(uint64_t numRows) {
     sharedErrorHandler->setHeaderNumRows(numRows);
 }
 
 LocalCSVFileErrorHandler::~LocalCSVFileErrorHandler() {
+    finalize();
+};
+
+void LocalCSVFileErrorHandler::finalize() {
     flushCachedErrors();
-}
+};
```

Comment:
Do we need `;` at the end? Ditto on L209.

Reply:
updated